openmc/mgxs_library.py

```diff
@@ -970,109 +1103,189 @@ def set_multiplicity_mgxs(self, nuscatter, scatter=None, nuclide='total',
+                for p in range(Np):
+                    for a in range(Na):
+                        for g_in in range(G):
+                            matrix = self._scatter_matrix[i][p, a, 0, g_in, :]
+                            nz = np.nonzero(matrix)
+                            g_out_bounds[p, a, g_in, 0] = nz[0][0]
+                            g_out_bounds[p, a, g_in, 1] = nz[0][-1]
+                # Now create the flattened scatter matrix array
+                flat_scatt = []
+                for p in range(Np):
+                    for a in range(Na):
+                        matrix = self._scatter_matrix[i][p, a, :, :, :]
+                        for g_in in range(G):
+                            for g_out in range(g_out_bounds[p, a, g_in, 0],
+                                               g_out_bounds[p, a, g_in, 1] + 1):
+                                for l in range(len(matrix[:, g_in, g_out])):
+                                    flat_scatt.append(matrix[l, g_in, g_out])
+                # And write it.
+                scatt_grp = xsgrp.create_group('scatter data')
+                scatt_grp.create_dataset("scatter matrix",
+                                         data=np.array(flat_scatt),
+                                         compression=compression)
+                # Repeat for multiplicity
+                if self._multiplicity_matrix[i] is not None:
+                    # Now create the flattened scatter matrix array
+                    flat_mult = []
+                    for p in range(Np):
+                        for a in range(Na):
+                            matrix = self._multiplicity_matrix[i][p, a, :, :]
+                            for g_in in range(G):
+                                for g_out in range(g_out_bounds[p, a, g_in, 0],
+                                                   g_out_bounds[p, a, g_in, 1] + 1):
+                                    flat_mult.append(matrix[g_in, g_out])
+                # And write it.
+                scatt_grp.create_dataset("multiplicity matrix",
+                                         data=np.array(flat_mult),
+                                         compression=compression)
+                # And finally, adjust g_out_bounds for 1-based group counting
+                # and write it.
+                g_out_bounds[:, :, :, :] += 1
+                scatt_grp.create_dataset("g_min", data=g_out_bounds[:, :, :, 0],
+                                         compression=compression)
+                scatt_grp.create_dataset("g_max", data=g_out_bounds[:, :, :, 1],
+                                         compression=compression)
 
 
 class MGXSLibrary(object):
     """Multi-Group Cross Sections file used for an OpenMC simulation.
     Corresponds directly to the MG version of the cross_sections.xml input
     file.
 
+    Parameters
+    ----------
+    energy_groups : openmc.mgxs.EnergyGroups
+        Energygroup structure
```

Comment:
Could you add a space between energy and group here and other places in this file?